parachain/pallets/incentivized-channel/src/inbound/mod.rs

```diff
@@ -106,3 +113,37 @@ decl_module! {
 		}
 	}
 }
+
+impl<T: Config> Module<T> {
+
+	fn handle_fee(amount: BalanceOf<T>, relayer: &T::AccountId) {
+		if amount.is_zero() {
+			return;
+		}
+
+		let imbalance = match T::Currency::withdraw(&T::SourceAccount::get(), amount, WithdrawReasons::TRANSFER, KeepAlive) {
```

Comment:
I don't think that actually has any effect at the moment for the balances pallet, looking at https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L532

The only reason that seems to change any withdrawal logic is `TRANSACTION_PAYMENT`. Presumably other implementations of `Currency` may do something different.

In any case, `T::SourceAccount` actually maps to the DOTApp account, rather than the account of a specific user. So I don't see any semantic benefit in marking the withdrawal as a fee. We're not really charging a fee from the DOTAccount in exchange for some action.

Reply:
I think the main difference is how it will show up in block explorer, if at all. A transfer has a sender and a recipient, while a fee is not "sent" to anyone. This is hypothetical though, because block explorers currently use events rather than withdraw reasons. 

No strong opinion, but to not mess up explorers, I guess changing to `FEE` might make sense.

Reply:
Hey yeah I remember we previously chatted about that. I agree. That's why I moved to withdraw & deposit instead of transfer, as events are only emitted for the latter (because its a high level operation). So we're fine in that regard.

Reply:
For reference, no events are emitted in `Currency::withdraw`: https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1473